lib/spack/spack/directives.py

```diff
@@ -561,14 +581,14 @@ def depends_on(spec, when=None, type=dt.DEFAULT_TYPES, patches=None):
 
     """
 
-    def _execute_depends_on(pkg):
+    def _execute_depends_on(pkg: "spack.package_base.PackageBase"):
         _depends_on(pkg, spec, when=when, type=type, patches=patches)
 
     return _execute_depends_on
 
 
 @directive(("extendees", "dependencies"))
-def extends(spec, type=("build", "run"), **kwargs):
+def extends(spec, when=None, type=("build", "run"), patches=None):
```

Comment:
(minor) Technically, if we want to be non-breaking, we need to swap the arguments:
```suggestion
def extends(spec, type=("build", "run"), when=None, patches=None):
```
However, I guess the number of users having a call to `extends` with only positional arguments is very low...

Reply:
I tend to agree. I don't think this is a major change.